### PR TITLE
CLI: Linter fixes

### PR DESCRIPTION
**Pull request by Lezek123**
Introduces fixes to all linter errors following https://github.com/Joystream/joystream/pull/917

### CODE REVIEW

**Review by mnaamani — CHANGES_REQUESTED** [review]
Great! 

Can you also run `yarn format` and check-in the changed files.
The reason the `--fix` argument when you ran linter didn't sort those out is because the `lint` script is only linting .ts files `/src` folder I think.

Perhaps update it the lint script to cover the whole directory. So `format`, `lint`, and `checks` all cover the same files.

Perhaps also update `.github/workflows/joystream-cli.yml`

to run `yarn workspace joystream-cli checks` in place of `yarn workspace joystream-cli build`

**Comment by mnaamani** [comment]
I think ideally

```
    "lint": "eslint . --quiet --ext .ts",
    "checks": "yarn lint && tsc --noEmit --pretty && prettier . --check",
    "format": "prettier . --write"
```

but will need to fix:

```
/Users/mokhtar/joystream/joystream/cli/test/commands/council/info.test.ts
  3:1  error  'describe' is not defined  no-undef

✖ 1 problem (1 error, 0 warnings)
```

I think adding mocha to the `env` in the eslintrc file should do the trick

**Comment by Lezek123** [comment]
Fixed the mentioned issues.
While testing I noticed the lack of unused imports check for the CLI.
I found this check very helpful in Pioneer. so I added `@typescript-eslint/no-unused-vars` rule to CLI's `.eslintrc.js` too. Also set `"noUnusedLocals": true` in `tsconfig.json` (the latter will additionaly cause the build to fail if there are any unused imports spotted).

Perhaps this can be part of the common standard?

File: .github/workflows/joystream-cli.yml [hunk 1]
```diff
@@ -14,14 +14,14 @@ jobs:
       uses: actions/setup-node@v1
       with:
         node-version: ${{ matrix.node-version }}
-    - name: build
+    - name: check
       run: |
         yarn install --frozen-lockfile
         yarn madge --circular types/
-        yarn workspace joystream-cli build
+        yarn workspace joystream-cli check
```
Comment (mnaamani) [inline .github/workflows/joystream-cli.yml]:
```suggestion
        yarn workspace joystream-cli checks
```

Its actually `checks`
`check` is a yarn [builtin command](https://classic.yarnpkg.com/en/docs/cli/check/#toc-yarn-check) that does sanity check on package.json and yarn.lock

**Review by mnaamani — COMMENTED** [review]
> Fixed the mentioned issues.
> While testing I noticed the lack of unused imports check for the CLI.
> I found this check very helpful in Pioneer. so I added `@typescript-eslint/no-unused-vars` rule to CLI's `.eslintrc.js` too. Also set `"noUnusedLocals": true` in `tsconfig.json` (the latter will additionaly cause the build to fail if there are any unused imports spotted).
> 
> Perhaps this can be part of the common standard?

Yes those rules look very good and can catch a tonne of bugs. I'm surprised the no-unused-vars isn't on by default in the rule-sets we have.

Added reminder in https://github.com/Joystream/joystream/issues/664

File: .github/workflows/joystream-cli.yml (hunk 1 above)
Comment (mnaamani) [reply .github/workflows/joystream-cli.yml]: sorry seems I missed the fix in later commit

**Pull request merged**